lib/dependencies/HarmonyCompatibilityDependency.js

```diff
@@ -3,20 +3,41 @@
 	Author Tobias Koppers @sokra
 */
 "use strict";
+
+/** @typedef {import('../Dependency').DependencyTemplate} DependencyTemplate **/
+/** @typedef {import('../Module')} Module **/
+/** @typedef {import("webpack-sources").ReplaceSource} ReplaceSource */
+/** @typedef {import("../RuntimeTemplate")} RuntimeTemplate */
+
 const NullDependency = require("./NullDependency");
 
 class HarmonyCompatibilityDependency extends NullDependency {
+	/**
+	 * @param {Module} originModule - original module
+	 */
 	constructor(originModule) {
 		super();
 		this.originModule = originModule;
 	}
 
+	/**
+	 * @override
+	 */
 	get type() {
 		return "harmony export header";
 	}
 }
 
+/**
+ * @implements {DependencyTemplate}
```

Comment:
If this `@implements` doesn't work, we should forbid the usage in `.eslintrc.js`.

Reply:
done in https://github.com/webpack/webpack/commit/a1aa5cbdce7110ab18aba098492853150a19aa0b